app/src/main/kotlin/com/wire/android/ui/home/conversations/model/MessageTypes.kt

```diff
@@ -322,22 +394,24 @@ private fun DownloadStatusIcon(assetDownloadStatus: Message.DownloadStatus) {
 }
 
 @Composable
-fun getDownloadStatusText(assetDownloadStatus: Message.DownloadStatus): String =
-    when (assetDownloadStatus) {
-        NOT_DOWNLOADED -> stringResource(R.string.asset_message_tap_to_download_text)
-        SAVED_INTERNALLY -> stringResource(R.string.asset_message_downloaded_internally_text)
-        IN_PROGRESS -> stringResource(R.string.asset_message_download_in_progress_text)
-        SAVED_EXTERNALLY -> stringResource(R.string.asset_message_saved_externally_text)
-        FAILED -> stringResource(R.string.asset_message_failed_download_text)
+fun getDownloadStatusText(assetDownloadStatus: Message.DownloadStatus, assetUploadStatus: Message.UploadStatus): String =
+    when {
+        assetUploadStatus == UPLOAD_IN_PROGRESS -> stringResource(R.string.asset_message_upload_in_progress_text)
+        assetUploadStatus == FAILED_UPLOAD -> stringResource(R.string.asset_message_failed_upload_text)
+        assetDownloadStatus == NOT_DOWNLOADED -> stringResource(R.string.asset_message_tap_to_download_text)
+        assetDownloadStatus == SAVED_INTERNALLY -> stringResource(R.string.asset_message_downloaded_internally_text)
+        assetDownloadStatus == DOWNLOAD_IN_PROGRESS -> stringResource(R.string.asset_message_download_in_progress_text)
+        assetDownloadStatus == SAVED_EXTERNALLY
+                || assetUploadStatus == UPLOADED -> stringResource(R.string.asset_message_saved_externally_text)
+        assetDownloadStatus == FAILED_DOWNLOAD -> stringResource(R.string.asset_message_failed_download_text)
+        else -> ""
     }
 
 @Suppress("MagicNumber")
-private fun provideAssetDescription(assetExtension: String, assetSizeInBytes: Long): String {
-    return when {
-        assetSizeInBytes < 1000 -> "${assetExtension.uppercase()} ($assetSizeInBytes B)"
-        assetSizeInBytes in 1000..999999 -> "${assetExtension.uppercase()} (${assetSizeInBytes / 1000} KB)"
-        else -> "${assetExtension.uppercase()} (${((assetSizeInBytes / 1000000f) * 100.0).roundToInt() / 100.0} MB)" // 2 decimals round off
-    }
+private fun provideAssetDescription(assetExtension: String, assetSizeInBytes: Long): String = when {
+    assetSizeInBytes < 1000 -> "${assetExtension.uppercase()} ($assetSizeInBytes B)"
+    assetSizeInBytes in 1000..999999 -> "${assetExtension.uppercase()} (${assetSizeInBytes / 1000} KB)"
```

Comment:
If I remember correctly, 1kB = 1000B as small k is a SI notation, but 1KB = 1024B because big K is used to represent the binary meaning 😄 
This only applies to kilos, mega and up are always capitalised, but probably nowadays also kilos are interchangeably especially as in the case of kilobytes the differences are not so big 😄